### PR TITLE
Update Docstrings, Change os.path to pathlib

### PR DESCRIPTION
**Pull request by ChihweiLHBird**

### CODE REVIEW

**Comment by ChihweiLHBird** [comment]
@rgerkin I think we can do `Squash and merge` option for this pull request because most commits messages are repeating. There are still many docstrings that need to be filled up, and I will try to do so recently. This pull request may be ready soon.

File: sciunit/validators.py [hunk 1]
```diff
@@ -33,8 +33,8 @@ class ObservationValidator(Validator):
     """Cerberus validator class for observations.
 
     Attributes:
-        test (Test): [Description of `test`].
-        _error (str, str): [Description of `_error`].
+        test (Test): A sciunit test instance to be validated.
+        _error (str, str): The error happends during the validating process.
```
[the remark this answers is not in the document]
Comment (rgerkin) [reply sciunit/validators.py:37]: Actually, where are these attributes even coming from?

File: sciunit/validators.py [hunk 2]
```diff
@@ -90,8 +90,8 @@ class ParametersValidator(Validator):
     """Cerberus validator class for observations.
 
     Attributes:
-        units_type ([type]): [description]
-        _error (str, str): [description]
+        units_type ([type]): The type of Python quantity's unit.
```
Comment (rgerkin) [inline sciunit/validators.py:93]: Where are these attributes coming from?

File: sciunit/validators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChihweiLHBird) [reply sciunit/validators.py:37]: These attributes are defined at line number 66, 80, and 85.

File: sciunit/validators.py (hunk 2 above)
Comment (ChihweiLHBird) [reply sciunit/validators.py:93]: `_error` is from line number 107 and 126.
`units_type` is from line number 119.
You can search `self.attribute_name` in the file to find where they are defined.

File: sciunit/suites.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ class TestSuite(SciUnit, TestWeighted):
     def __init__(self, tests: List[Test], name: str=None, weights=None, include_models: List[Model]=None,
                  skip_models: List[Model]=None, hooks: dict=None, 
                  optimizer=None):
-        """optimizer: a function to bind to self.optimize (first argument must be a testsuite).
+        """The constructor of TestSuite class.
```
[the remark this answers is not in the document]
Comment (ChihweiLHBird) [reply sciunit/suites.py:24]: Maybe we can do that, I did not find a strict rule about the docstring for constructor. Some Google's constructors even don't have a docstring.

File: sciunit/utils.py [hunk 4]
```diff
@@ -94,45 +96,26 @@ def dict_combine(*dict_list) -> dict:
...
+        state (bool): A state of an instance.
     """
 
-    global settings
-    settings['PRINT_DEBUG_STATE'] = (state is True)
+    config_set('PRINT_DEBUG_STATE', (state is True))
 
 
 def printd(*args, **kwargs) -> bool:
     """Print if PRINT_DEBUG_STATE is True.
```
[the remark this answers is not in the document]
Comment (ChihweiLHBird) [reply sciunit/utils.py:112]: I am sorry but I still don't understand...
It seems means to set 'PRINT_DEBUG_STATE' if `state` is `True`, and I did not change its logic. But I do think `(state is True)` is equivalent to `state`. And it seems doesn't print anything.

**Pull request merged**